packages/muelu/test/contact/SegregatedAggFromFactoryBlockedA.cpp

```diff
@@ -0,0 +1,371 @@
+
+  const std::string matrixFileName             = probName + "_block_matrix.mm";
+  const std::string nullspace1FileName         = probName + "_nullspace.mm";
+  const std::string primalInterfaceMapFileName = probName + "_interface_dof_map_MPI" + std::to_string(numRanks) + ".mm";
+  const std::string dropMap1Name               = "dropMap1";
+  const std::string dropMap2Name               = "dropMap2";
+  const std::string droppingScheme             = "map-pair";
+
+  /// Rowmaps
+  // Primal dofmap
+  std::vector<size_t> stridingInfoPrimal;
+  stridingInfoPrimal.push_back(numPrimalDofPerNode);
+  RCP<const StridedMap> StridedDofRowMapPrimal = StridedMapFactory::Build(lib, numGlobalDofPrimal,
+                                                                          Teuchos::ScalarTraits<GO>::zero(),
+                                                                          stridingInfoPrimal, comm, -1);
+
+  // Dual dofmap
+  std::vector<size_t> stridingInfoDual;
+  stridingInfoDual.push_back(numDualDofPerNode);
+  RCP<const StridedMap> StridedDofRowMapDual = StridedMapFactory::Build(lib, numGlobalDofDual,
+                                                                        Teuchos::ScalarTraits<GO>::zero(),
+                                                                        stridingInfoDual, comm, -1, numGlobalDofPrimal);
+
+  // Construct the blocked map of the global system
+  std::vector<RCP<const Map>> rowmaps;
+  rowmaps.push_back(StridedDofRowMapPrimal);
+  rowmaps.push_back(StridedDofRowMapDual);
+  RCP<const Map> fullRowMap        = MapUtils::concatenateMaps(rowmaps);
+  RCP<const BlockedMap> blockedMap = rcp(new BlockedMap(fullRowMap, rowmaps));
+
+  /// Matrix A
+  // Read the matrix from file and transform it into a block matrix
+  RCP<Matrix> mat                     = Xpetra::IO<SC, LO, GO, NO>::Read(matrixFileName, fullRowMap);
+  RCP<MapExtractor> rangeMapExtractor = Xpetra::MapExtractorFactory<SC, LO, GO, NO>::Build(fullRowMap, rowmaps);
+  RCP<BlockedCrsMatrix> blockedMatrix = Xpetra::MatrixUtils<SC, LO, GO, NO>::SplitMatrix(*mat, rangeMapExtractor,
+                                                                                         rangeMapExtractor);
+
+  blockedMatrix->fillComplete();
+
+  /// Nullspaces
+  // Read the nullspace vector of the (0,0)-block from file
+  RCP<MultiVector> nullspace1 = Xpetra::IO<SC, LO, GO, NO>::ReadMultiVector(nullspace1FileName, StridedDofRowMapPrimal);
+
+  // Create the default nullspace vector of the (1,1)-block
+  RCP<MultiVector> nullspace2 = MultiVectorFactory::Build(StridedDofRowMapDual, 2, true);
+  const int dimNS             = 2;
+  for (int dim = 0; dim < dimNS; ++dim) {
+    ArrayRCP<Scalar> nsValues = nullspace2->getDataNonConst(dim);
+    const int numBlocks       = nsValues.size() / dimNS;
+    for (int j = 0; j < numBlocks; ++j)
+      nsValues[j * dimNS + dim] = Teuchos::ScalarTraits<Scalar>::one();
+  }
```

Comment:
Is this different from what a nullspace factory would produce?

Reply:
No, it should be the same as the output from the nullspace factory. However, in our internal research code this is how we generate the nullspace before we pass it to mueLu, as other applications than contact might differ.  I wanted to replicate this workflow in the test (which is similarly done in the meshtying tests).